ecs-web/main.tf

```diff
@@ -46,70 +47,78 @@ resource "aws_ecs_task_definition" "concourse_web_task_definition" {
+    vault_command_args        = var.vault_server_url != null ? "--vault-client-token=`cat /concourse_vault/token`" : ""
+    vault_auth_task_definition = indent(
+      2,
+      join("", data.template_file.vault_auth_task_template.*.rendered),
+    )
   }
 }
 
 data "template_file" "concourse_db_task_template" {
-  count    = "${var.auto_create_db ? 1 : 0}"
-  template = "${file("${path.module}/task-definitions/create_concourse_db_container.json")}"
+  count = var.auto_create_db ? 1 : 0
+  template = file(
+    "${path.module}/task-definitions/create_concourse_db_container.json",
+  )
 
-  vars {
+  vars = {
     image                      = "postgres"
-    image_tag                  = "${var.concourse_db_postgres_engine_version == "" ? "latest" : var.concourse_db_postgres_engine_version}"
-    concourse_db_host          = "${var.concourse_db_host}"
-    concourse_db_port          = "${var.concourse_db_port}"
-    concourse_db_user          = "${var.concourse_db_username}"
-    concourse_db_password      = "${jsonencode(var.concourse_db_password)}"
-    concourse_db_name          = "${var.concourse_db_name}"
-    concourse_db_root_password = "${jsonencode(var.concourse_db_root_password)}"
-    awslog_group_name          = "${aws_cloudwatch_log_group.concourse_web_log_group.name}"
-    awslog_region              = "${data.aws_region.current.name}"
+    image_tag                  = coalesce(var.concourse_db_postgres_engine_version, "latest")
+    concourse_db_host          = var.concourse_db_host
+    concourse_db_port          = var.concourse_db_port
+    concourse_db_user          = var.concourse_db_username
+    concourse_db_password      = jsonencode(var.concourse_db_password)
+    concourse_db_name          = var.concourse_db_name
+    concourse_db_root_password = jsonencode(var.concourse_db_root_password)
+    awslog_group_name          = aws_cloudwatch_log_group.concourse_web_log_group.name
+    awslog_region              = data.aws_region.current.name
   }
 }
 
 data "template_file" "vault_auth_task_template" {
-  count    = "${length(var.vault_server_url) > 0 ? 1 : 0}"
-  template = "${file("${path.module}/task-definitions/vault_auth_container.json")}"
+  count    = var.vault_server_url !=null ? 1 : 0
+  template = file("${path.module}/task-definitions/vault_auth_container.json")
 
-  vars {
+  vars = {
     image             = "vault"
-    image_tag         = "${var.vault_docker_image_tag}"
-    vault_addr        = "${var.vault_server_url}"
-    auth_header_value = "${replace(replace(var.vault_server_url, "/^http(s)?:///", ""), "/", "")}"
-    auth_role         = "${var.vault_auth_concourse_role_name}"
-    awslog_group_name = "${aws_cloudwatch_log_group.concourse_web_log_group.name}"
-    awslog_region     = "${data.aws_region.current.name}"
+    image_tag         = var.vault_docker_image_tag
+    vault_addr        = var.vault_server_url
+    auth_header_value = replace(replace(var.vault_server_url, "/^http(s)?:///", ""), "/", "")
```

Comment:
Not sure about it, but this might give an error if `var.vault_server_url` is `null` 😅

Reply:
no didn't give error during my test, I guess it's because there's a count for the data resource and it's used within a join function so it's just ok to be null.